assets/js/components/DashboardEntryPoint.test.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * DashboardEntryPoint tests.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { render } from '../../../tests/js/test-utils';
+import { mockCreateComponent } from '../../../tests/js/mock-component-utils';
+import DashboardEntryPoint from './DashboardEntryPoint';
+
+jest.mock( './setup/ModuleSetup', () => mockCreateComponent( 'ModuleSetup' ) );
+jest.mock( './dashboard/DashboardApp', () =>
+	mockCreateComponent( 'DashboardApp' )
+);
+jest.mock( './DashboardMainApp', () =>
+	mockCreateComponent( 'DashboardMainApp' )
+);
+jest.mock( './legacy-notifications/notification-counter', () =>
+	mockCreateComponent( 'NotificationCounter' )
+);
```

Comment:
If we move these `jest.mock` calls to a `beforeAll` block, the tests won't work as they stand. This is because they are relying on the hoisting behaviour of `jest.mock` which means the calls to `jest.mock` are hoisted above the `import` of `DashboardEntryPoint`.

In order for the mocking to work within a `beforeAll` we'd need to do something like this, with a `require` after the mocks:

```ts
	let DashboardEntryPoint;

	beforeAll( () => {
		jest.mock( './setup/ModuleSetup', () =>
			mockCreateComponent( 'ModuleSetup' )
		);
		jest.mock( './dashboard/DashboardApp', () =>
			mockCreateComponent( 'DashboardApp' )
		);
		jest.mock( './DashboardMainApp', () =>
			mockCreateComponent( 'DashboardMainApp' )
		);
		jest.mock( './legacy-notifications/notification-counter', () =>
			mockCreateComponent( 'NotificationCounter' )
		);

		DashboardEntryPoint = require( './DashboardEntryPoint' ).default;
	} );
```

Personally, I prefer to keep the `jest.mock` calls at the outermost scope to reduce the amount of boilerplate, but I don't mind if you still would rather move them to a `beforeAll`. Please let me know what you think. Cheers!

Reply:
Ah... ok, thanks for letting me know. Works for me then 👍 .